blazor/datagrid/paging.md

```diff
@@ -7,443 +7,1162 @@ control: DataGrid
+        decimal PageResize = ((GridHeight) - (PageSize * RowHeight)) / RowHeight; //new page size is obtained here
+        Grid.PageSettings.PageSize = PageSize + (int)Math.Round(PageResize);
+        await Grid.Refresh();
     }
-    public void LaunchEnteredPage(Microsoft.AspNetCore.Components.ChangeEventArgs page)
+}
+{% endhighlight %}
+{% highlight c# tabtitle="OrderData.cs" %}
+ public class OrderData
     {
-        if (page.Value == null || page.Value.ToString() == "")
+        public static List<OrderData> Orders = new List<OrderData>();
+        
+        public OrderData()
         {
-            return;
+
         }
-        else
+        public OrderData(int? OrderID,string CustomerID, DateTime? OrderDate, double? Freight)
         {
-            double enteredPage = Convert.ToDouble(page.Value);
-            if (enteredPage <= totalPages)
-                defaultGrid.GoToPageAsync(enteredPage);
+           this.OrderID = OrderID;    
+           this.CustomerID = CustomerID;
+            this.OrderDate = OrderDate;
+            this.Freight = Freight;            
         }
+
+        public static List<OrderData> GetAllRecords()
+        {
+            if (Orders.Count() == 0)
+            {
+                int OrderID = 10248;
+                for (int i = 1; i < 7; i++)
+                {
+                    Orders.Add(new OrderData(OrderID+1, "VINET", new DateTime(1996, 07, 06), 32.38));
+                    Orders.Add(new OrderData(OrderID+2, "TOMSP", new DateTime(1996, 07, 06), 11.61));
+                    Orders.Add(new OrderData(OrderID+3, "HANAR", new DateTime(1996, 07, 06), 65.83));
+                    Orders.Add(new OrderData(OrderID+4, "VICTE", new DateTime(1996, 07, 06), 45.78));
+                    Orders.Add(new OrderData(OrderID+5, "SUPRD", new DateTime(1996, 07, 06), 98.6));
+                    Orders.Add(new OrderData(OrderID+6, "HANAR", new DateTime(1996, 07, 06), 103.45));
+                    Orders.Add(new OrderData(OrderID+7, "CHOPS", new DateTime(1996, 07, 06), 103.45));
+                    Orders.Add(new OrderData(OrderID+8, "RICSU", new DateTime(1996, 07, 06), 112.48));
+                    Orders.Add(new OrderData(OrderID+9, "WELLI", new DateTime(1996, 07, 06), 33.45));
+                    OrderID += 9;
+                }
+            }
+            return Orders;
+        }
+
+        public int? OrderID { get; set; }
+        public string CustomerID { get; set; }
+        public DateTime? OrderDate { get; set; }
+        public double? Freight { get; set; }
     }
-}
-```
+{% endhighlight %}
+{% endtabs %}
 
-N> You can refer to our [Blazor Grid Pager Template](https://blazor.syncfusion.com/demos/datagrid/pager-template) online demo of Pager Template feature in Blazor DataGrid.
+{% previewsample "https://blazorplayground.syncfusion.com/embed/rZhfXiiWLoaAoCIo?appbar=false&editor=false&result=true&errorlist=false&theme=bootstrap5" %}
```

Comment:
initially give empty value as in angular doc

Reply:
Resolved